demo/runtime/src/runtime/council.rs

```diff
@@ -0,0 +1,1236 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Substrate Demo.
+
+// Substrate Demo is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate Demo is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate Demo.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Council system: Handles the voting in and maintenance of council members.
+
+use rstd::prelude::*;
+use codec::KeyedVec;
+use runtime_support::storage;
+use demo_primitives::{Proposal, AccountId, Hash, BlockNumber};
+use runtime::{staking, system, session};
+use runtime::staking::Balance;
+
+// no polynomial attack:
+// all public operations should be constant time.
+// all protected operations may be at most O(public operations)
+
+// public operations:
+// - express approval (ideally, express all approvals)
+// - re-express approval (ideally, re-express all approvals)
+// - clear all approvals
+// - submit candidacy
+// protected operations:
+// - remove candidacy (remove all votes for a candidate)
+
+// for an approval vote of C councilers:
+
+// top K candidates are maintained between votes. all others are discarded.
+// - candidate removed & bond returned when elected.
+// - candidate removed & bond burned when discarded.
+
+// at the point that the vote ends, all voters' balances are snapshotted.
+
+// for B blocks following, there's a counting period whereby each of the candidates that believe
+// they fall in the top K+C voted can present themselves. they get the total stake
+// recorded (based on the snapshot); an ordered list is maintained. Noone may present themselves
+// that, if elected, would result in being included twice on the council (important since existing
+// councilers will will have their approval votes as it may be that they don't get removed), nor
+// if existing presenters would mean they're not in the top K+C.
+
+// following B blocks, the top C+K that presented themselves have their bond returned and the
+// top C candidates are elected. the top C candidates have their bond returned.
```

Comment:
is it the top C+K or the top C that have their bond returned?

Reply:
and everyone who loses the election loses their money?

Reply:
everyone who outright loses it, yes. those who come close enough (within `carry_count`) of winning get to stick around for the next vote.